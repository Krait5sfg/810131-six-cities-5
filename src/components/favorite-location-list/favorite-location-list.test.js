import React from 'react';
import renderer from 'react-test-renderer';
import FavoriteLocationList from './favorite-location-list';

const offers = [
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [],
      guestsLimit: 0,
      isPremium: true,
      price: 0,
      rating: 0,
      title: ``,
      type: ``,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [0, 0],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 1,
    images: [],
    isFavorite: false,
    previewImage: ``,
  },
  {
    city: ``,
    accommodation: {
      bedroomsCount: 0,
      features: [``],
      guestsLimit: 0,
      isPremium: false,
      price: 0,
      rating: 0,
      title: ``,
      type: ``,
    },
    cityLocation: {latitude: 0, longitude: 0, zoom: 0},
    coordinates: [0, 0],
    description: ``,
    host: {
      avatar: ``,
      isPro: true,
      name: ``,
    },
    id: 0,
    images: [],
    isFavorite: true,
    previewImage: ``,
  },
];

jest.mock(`../favorite-place-card/favorite-place-card`, () => `FavoritePlaceCard`);

it(`FavoriteLocationList render correctly`, () => {
  const tree = renderer.create(<FavoriteLocationList
    city={`Paris`}
    cityFavoriteOffers={offers}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
