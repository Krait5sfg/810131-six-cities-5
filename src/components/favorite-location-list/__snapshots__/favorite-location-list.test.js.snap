// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteLocationList render correctly 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <a
        className="locations__item-link"
      >
        <span>
          Paris
        </span>
      </a>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <FavoritePlaceCard
      favoriteOffer={
        Object {
          "accommodation": Object {
            "bedroomsCount": 0,
            "features": Array [],
            "guestsLimit": 0,
            "isPremium": true,
            "price": 0,
            "rating": 0,
            "title": "",
            "type": "",
          },
          "city": "",
          "cityLocation": Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 0,
          },
          "coordinates": Array [
            0,
            0,
          ],
          "description": "",
          "host": Object {
            "avatar": "",
            "isPro": true,
            "name": "",
          },
          "id": 1,
          "images": Array [],
          "isFavorite": false,
          "previewImage": "",
        }
      }
    />
    <FavoritePlaceCard
      favoriteOffer={
        Object {
          "accommodation": Object {
            "bedroomsCount": 0,
            "features": Array [
              "",
            ],
            "guestsLimit": 0,
            "isPremium": false,
            "price": 0,
            "rating": 0,
            "title": "",
            "type": "",
          },
          "city": "",
          "cityLocation": Object {
            "latitude": 0,
            "longitude": 0,
            "zoom": 0,
          },
          "coordinates": Array [
            0,
            0,
          ],
          "description": "",
          "host": Object {
            "avatar": "",
            "isPro": true,
            "name": "",
          },
          "id": 0,
          "images": Array [],
          "isFavorite": true,
          "previewImage": "",
        }
      }
    />
  </div>
</li>
`;
